### PR TITLE
Merge/develop ver 3 to develop

### LINKED ISSUE #1949: DatasetApiIntegrationSuite failing test on merge to Enceladus 3
## Describe the bug
As seen in [here](https://github.com/AbsaOSS/enceladus/pull/1928/files/580cc478177de69ec6a65a242cd2c556b081d34d#r719432243) the test is failing on an empty map, which I suspect is caused by the test client

### LINKED ISSUE #2070: Merge long living `develop-ver-3.0` branch into `develop`
## Background
Enceladus preparations for Release ver. 3.0.0

## Feature
The long living code branch `develop-ver-3.0` is not needed anymore. Merge it into `develop` and delete it.

### PR DESCRIPTION
**Pull request by AdrianOlosutean**
Closes #2070 
Closes #1949 
Looks like the test works again

### CODE REVIEW

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/services/DatasetServiceTest.scala [hunk 1]
```diff
@@ -345,10 +346,10 @@ class DatasetServiceTest extends VersionedModelServiceTest[Dataset] with Matcher
     Mockito.when(modelRepository.getVersion("datasetA", 1)).thenReturn(Future.successful(Some(dataset)))
     Mockito.when(modelRepository.getLatestVersionValue("datasetA")).thenReturn(Future.successful(Some(1)))
 
-    val maybeDataset = await(service.update("datasetA",
+    val maybeDataset: Option[(Dataset, Validation)] = await(service.update("datasetA",
       DatasetFactory.getDummyDataset(name = "datasetA", description = Some("newdescr"))))
-    assertResult(Some("newdescr"))(maybeDataset.get.description)
-    assert(!maybeDataset.get.lockedWithDefault)
+    assertResult(Some("newdescr"))(maybeDataset.get._1.description)
```
Comment (AdrianOlosutean) [inline rest-api/src/test/scala/za/co/absa/enceladus/rest_api/services/DatasetServiceTest.scala:351]: I added these changes manually. Maybe if @dk1844 can double check that the code matches his

File: rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/SchemaService.scala [hunk 2]
```diff
@@ -56,19 +56,33 @@ class SchemaService @Autowired() (schemaMongoRepository: SchemaMongoRepository,
     } yield update
   }
 
-  override def update(username: String, schema: Schema): Future[Option[Schema]] = {
-    super.update(username, schema.name, schema.version) { latest =>
-      latest.setDescription(schema.description).asInstanceOf[Schema]
+  /**
```
Comment (AdrianOlosutean) [inline rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/SchemaService.scala:59]: I think this file also deserve a double check

File: rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/DatasetService.scala [hunk 3]
```diff
@@ -427,12 +438,22 @@ object DatasetService {
...
    */
-  def removeBlankProperties(properties: Option[Map[String, String]]): Option[Map[String, String]]  = {
+  private[services] def removeBlankPropertiesOpt(properties: Option[Map[String, String]]): Option[Map[String, String]]  = {
     properties.map {
-      _.filter { case (_, propValue) => propValue.nonEmpty }
+      removeBlankProperties
     }
   }
 
+  /**
```
Comment (AdrianOlosutean) [inline rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/DatasetService.scala:447]: This file may need checking too

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/controllers/DatasetApiIntegrationSuite.scala [hunk 4]
```diff
@@ -290,7 +291,7 @@ class DatasetApiIntegrationSuite extends BaseRestApiTest with BeforeAndAfterAll
 
           assert(headers2.getFirst("Location").contains("/api/dataset/dataset/2"))
           assert(body2.version == 2)
-          assert(body2.properties == expectedPropertiesSet)
+          //TODO DatasetApiIntegrationSuite failing test on merge to Enceladus 3 #1949
```
[the remark this answers is not in the document]
Comment (AdrianOlosutean) [reply rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/controllers/DatasetApiIntegrationSuite.scala]: It looks like the issue was solved somehow

**Comment by benedeki** [comment]
Nice on fixing #1949 too. (Despite it's still not clear why it was failing 😉 )

File: rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/DatasetService.scala [hunk 5]
```diff
@@ -13,22 +13,24 @@
  * limitations under the License.
  */
 
-package za.co.absa.enceladus.menas.services
+package za.co.absa.enceladus.rest_api.services
 
 import javax.ws.rs.NotAllowedException
```
Comment (benedeki) [inline rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/DatasetService.scala:18]: Unused

**Review by dk1844 — APPROVED** [review]

**Review by benedeki — APPROVED** [review]
- [X] code reviewed
- [X] pulled
- [X] built
- [ ] run
🤞

**Pull request merged**